CONTRIBUTING.md

```diff
@@ -6,23 +6,119 @@ JupyterHub also follows the Jupyter [Community Guides](https://jupyter.readthedo
 
 ## Set up your development system
 
-For a development install, clone the [repository](https://github.com/jupyterhub/jupyterhub)
-and then install from source:
+### System requirements
 
-```bash
-git clone https://github.com/jupyterhub/jupyterhub
-cd jupyterhub
-npm install -g configurable-http-proxy
-pip3 install -r dev-requirements.txt -e .
-```
+JupyterHub can only run on MacOS or Linux operating systems. If you
+are using Windows, we recommend using [VirtualBox](https://www.virtualbox.org/)
+or a similar system to run [Ubuntu Linux](https://www.ubuntu.com/)
+for development.
+
+### Install Python
+
+JupyterHub is written in the [Python](https://python3.org) programming
+language, and requires you have at least version 3.5 installed locally.
+If you haven't installed Python before, the recommended way to install
+it is to use [miniconda](https://conda.io/miniconda.html). Remember
+to get the 'Python 3' version, and **not** the 'Python 2' version!
+
+### Install nodejs
+
+`configurable-http-proxy`, the default proxy implementation for JupyterHub,
+is written in Javascript to run on [NodeJS](https://nodejs.org/en/). If
+you have not installed nodejs before, we recommend installing it in the
+`miniconda` environment you set up for Python. You can do so with
+`conda install nodejs`.
+
+### Install git
+
+JupyterHub uses [git](https://git-scm.com) & [GitHub](https://github.com)
+for development & collaboration. You need to [install git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+to work on JupyterHub. We also recommend getting a free account
+on GitHub.com.
+
+## Set up a development install
+
+When developing JupyterHub, you need to make changes to the code
+& see their effects quickly. You need to do a developer install
+to make that happen.
```

Comment:
This is awesome, you write excellent documentation.

Reply:
Thank you, this makes me feel very happy & appreciated <3